packages/studio-base/src/panels/ThreeDeeRender/SceneExtension.ts

```diff
@@ -80,6 +80,13 @@ export class SceneExtension<
     this.renderables.clear();
   }
 
+  /**
+   * Will add subscriptions from this scene extension to the renderer
+   * This will be called by the renderer when building topic and schema subscriptions on
+   * initialization and when imageOnlyMode becomes enabled
+   */
+  public addSubscriptionsToRenderer(): void {}
```

Comment:
I think this method shouldn't mutate anything on renderer, but should return a list of subscriptions. Then the renderer can add them as it sees fit.
Something more like this:

```suggestion
  type Subscription = { type: "topic", topic: string, subscription: RendererSubscription<T> | MessageHandler<T> } | { type: "schemas", schemaNames: Iterable<string>, subscription: RendererSubscription<T> | MessageHandler<T> };
  public getSubscriptions(): Subscription[] {}
```

Reply:
I know, I'm not happy with it either. that's what I originally wanted to do, but couldn't figure out how to safely type the Subscriptions without casting all of the handlers' messageEvent inputs to `unknown`

Reply:
Ah yeah, I mean realistically, if we want them in an array, TS doesn't have the ability to provide _separate_ strong types for all the items individually. I think we probably have this issue already with the renderer's internal storage of the list of subscriptions.

I'll file a separate followup ticket for myself to come back and look at alternatives to this after it merges